lib/samples/show_service_area/show_service_area_sample.dart

```diff
@@ -25,139 +25,234 @@ class ShowServiceAreaSample extends StatefulWidget {
-    color: Colors.white,
-    style: SimpleMarkerSymbolStyle.circle,
-    size: 10,
-  )..outline = SimpleLineSymbol(
-      style: SimpleLineSymbolStyle.solid,
-      color: Colors.black,
-      width: 2,
-    );
+  // create graphics overlays for displaying facilities and barriers.
+  final _facilityGraphicsOverlay = GraphicsOverlay();
+  final _barrierGraphicsOverlay = GraphicsOverlay();
 
-  final _serviceAreaSymbol = SimpleFillSymbol(
-    style: SimpleFillSymbolStyle.solid,
-    color: const Color(0x40FF3232),
-    outline: SimpleLineSymbol(
-      style: SimpleLineSymbolStyle.solid,
-      color: Colors.black,
-      width: 1,
+  // create a graphics overlay for displays service area results and a list of symbols applied for each impedence cutoff added to the service area parameters.
+  final _serviceAreaGraphicsOverlay = GraphicsOverlay();
+  final _serviceAreaSymbols = [
+    SimpleFillSymbol(
+      color: Colors.yellowAccent.withOpacity(0.4),
+      outline: SimpleLineSymbol(color: Colors.yellow.withOpacity(0.8)),
+    ),
+    SimpleFillSymbol(
+      color: Colors.orangeAccent.withOpacity(0.4),
+      outline: SimpleLineSymbol(color: Colors.orange.withOpacity(0.8)),
     ),
-  );
+    SimpleFillSymbol(
+      color: Colors.greenAccent.withOpacity(0.4),
+      outline: SimpleLineSymbol(color: Colors.green.withOpacity(0.8)),
+    ),
+  ];
+
+  // create a service area task used to find service areas around a facility.
+  final _serviceAreaTask = ServiceAreaTask.withUrl(Uri.parse(
+      'https://route-api.arcgis.com/arcgis/rest/services/World/ServiceAreas/NAServer/ServiceArea_World'));
+  late final ServiceAreaParameters _serviceAreaParameters;
+
+  var _ready = false;
+  var _segmentedButtonSelection = Selection.facility;
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       body: SafeArea(
-        child: Column(
+        top: false,
+        child: Stack(
           children: [
-            Expanded(
-              child: ArcGISMapView(
-                controllerProvider: () => _mapViewController,
-                onMapViewReady: onMapViewReady,
-                onTap: _isInitialized ? onTap : null,
-              ),
+            Column(
+              children: [
+                Expanded(
+                  // add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                    onTap: _ready ? onTap : null,
+                  ),
+                ),
+                SizedBox(
+                  height: 60,
+                  child: Row(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      // create segmented buttons for toggling adding a facility or barrier to the map.
+                      SegmentedButton<Selection>(
```

Comment:
Oops I forgot to mention -- I *think* it's smart enough to infer the type, so this can just be `SegmentedButton(`